core/trino-main/src/main/java/io/trino/server/HttpRequestSessionContextFactory.java

```diff
@@ -154,6 +154,14 @@ else if (nameParts.size() == 2) {
         catalogSessionProperties = catalogSessionProperties.entrySet().stream()
                 .collect(toImmutableMap(Entry::getKey, entry -> ImmutableMap.copyOf(entry.getValue())));
 
+        Optional<String> optionalAuthorizationUser = Optional
```

Comment:
the commit message reads like (atleast to me) - we're just adding a field for propagation. But this contains is a security related change - because it allows the value provided in  `X-Trino-Authorization-User` to be used for query execution after going through appropriate access control checks. 


IMO it'd be better to introduce the actual `SET SESSION AUTHORIZATION` change in the same PR as a separate commit - rather than having separate PRs. It will make it easier to see (for readers/reviewers ) why this header is needed by the server - especially when there's already a similar looking header `X-Trino-User`. We'll also avoid potentially cases where some trino versions will support this header without supporting the `SET SESSION AUTHORIZATION` command.

Reply:
The execution of set session authorization task requires the information of the original identity and that's the reason why I put this PR as the first one in this feature. I think it may make sense to have the current PR be separated as the `SET SESSION AUTHORIZATION` one. This PR adds a new header `X-Trino-Authorization-User`, and the server will do the impersonation check and execute the query as the authorization user if the header exists in the request. So this PR alone can support impersonation. I agree the commit message is misleading and I will update it. The follow-up PR for adding `SET SESSION AUTHORZATION` command can be considered as adding an easier way for users to set `X-Trino-Authorization-User` header in the query requests.

Reply:
Updated the PR title and description.

Reply:
Fixed